WeakAuras/GenericTrigger.lua

```diff
@@ -756,8 +758,15 @@ function WeakAuras.ScanEventsInternal(event_list, event, arg1, arg2, ... )
     local updateTriggerState = false;
     for triggernum, data in pairs(triggers) do
       local allStates = WeakAuras.GetTriggerStateForTrigger(id, triggernum);
-      if (RunTriggerFunc(allStates, data, id, triggernum, event, arg1, arg2, ...)) then
-        updateTriggerState = true;
+      local delay = GenericTrigger.GetDelay(data)
+      if delay == 0 then
+        if (RunTriggerFunc(allStates, data, id, triggernum, event, arg1, arg2, ...)) then
+          if GenericTrigger.MatchEventCount(data, id, triggernum) then
```

Comment:
This fundamentally can't work. The trigger has already updated the state, and you are just hiding that the state was changed from immediate further processing, here. But that doesn't make that state change invisible to a lot of other systems.

This will break e.g. conditions, but also multiple triggers. 
For example take this aura:
!WA:2!LrvZUTTrq4i4w0c6lXQXgnO)zKGgKEjnrbjh6nrfjBxilPqrh7CPsljhrU2u7YU7sjlFlQab5SEeeqV1t6rWpblekYdGFe8tqNDPKDAlpqm7SZoZ333m7wQr5HLJkh9HhoNgYzD55Iq4BVJ482dgibvjXKvgliSWeUOdNYub1Q3YVU3YqopnIpM1Dmnd6F56L1JIHext66pvjOXXGq(BpsSY8JUQjzWSq(WaIkLhplxc9c55mv)RK5bWiGP6Mpya98f9QvTRFVU(v98NgLlikkN9GN)upBmZRzZWUn5XEmYqq64MZOQGSuYeq4zZyLN9Zp)kt(LwEDeU)TfPJaWI41Tt9MnNkZG00dIKoZVnu92ZJGaek(iGf7xVzNgh1CgM2cI48iTZ713tVTEZGWCPcHJnVYST70S6BR7vG86VQ3HTFZbT2Brrq9m0xVL(70FV(ULFUE3SVzqol0WTh)to7kavUGTRsKdoalA9HsOrGNIoeI0)GZIiQ8u7zgbviSjzBrS2(fa7qEe8x35ASrbcgj9nO8Jj)pU(maYQIenu5zKYKzsiDGTHQ)CKPsIYOXqYCuH8Gy8mYFzh9MlXLbKWZIfOKg9HTZkRGZv9KjeSzFYQXJS7BDA)nGlgsWawz4Y4my2nBF5p(SNyv7wytl7UFsUQXt5IFTuPsBCvH31WZT6r(TZ(kRtsokhi8dpMgPsCRIlVUOQ0ZHiRZ3TSyw9TRW2L29pffs6GjivfCfstF0PBR2TQR3uGm0eZCBGHwySb(nBfDyQS72qqVy3xNtIqjISRV))sfwxPLwFJ5IOJfKSPhVYOaaJOsAqk0FfN)K7tzBDqRw48IBBF)2h6DWE77V8MsBh(AFKFZdAv)wVDPxa3FfpVPh8W)xp4tK6vU0v(4Y4u(4gc43Zbw4KoF2tFsLxuicMnwy(ToLtnlmiyzqUsXzThbc8(1CJ7MalwLSTn(1cWmZIA3iGxI392BTNexZMj2Q7NqdpJbs5g2t0nKKcfMnPO7TSfWfLpqKeKa04e17Ct5KOgbkmuMAhVH5PkQJRzEEhDphVWuIuASCLO2ymMlSdXg4R)YS7T(5jtJ)vujb7frjZim6q7dlV0tQicvJlbIe6Qew29f6n1vwU(Th7vxHeWx2IKtnXzZDf3HekRHo2gTort1NQRemGYOYK)R7fuwrJaZNJ3y78Q7fC(WscI9ra5l1dC0PoU4XvoZv8WrfxG)6sV)pFanQslyS3TZOlgiWMlIxCSCJRlgiAyCzGwq3AE1R3QsonAXRPVO2jp(qy)97eGeyan2rqzMCdjzLXgvTv6d(8b((a03JKMLq2yUHTulYCMUwd7VeVhIvQDwXgLLvEfmQ8O)(K)5d

The first trigger is a combat log trigger that shows every 3. spell cast for 30s
The second trigger is a custom "PLAYER_STARTED_MOVING" trigger.

Now, if you just start casting random spell it looks like everything is working correctly, that is the information of every 3 spell cast even is shown.

But the "hidden" state is easily exposed via:
1) Start casting spell foo
2) Start casting spell bar
3) Start moving

=> the aura shows the information for spell "bar".

That same problem exists for lots of systems, just hiding that a state has changed won't work.

Now you might be able to add the counting check to the trigger by modifying GenericTrigger's ConstructFunction.
The end of the function that is generated there is essentially
```
if ( **tests** ) then
   create state
   return true
else
  return false
end
```

Whereas if I see that correctly, you'd "just" need to modify that too:
```
if ( **tests** ) then
   if (use_count) then
     count = count + 1
     if not check_count then
       return false
     end
   end     
   create state
   return true
else
  return false
end
```

The other option would be some "watching" trigger that counts the number of times the watched trigger activates, but wathching triggers have a few usuability challenges, which makes me reject that idea.